pages/admin/invite/userAccountPage/[id].tsx

```diff
@@ -0,0 +1,166 @@
+import DashboardLayout from "components/DashboardLayout";
+// import Combobox from "components/Combobox";
+import React, { useState, useEffect } from "react";
+// import { User } from "@prisma/client";
+import { useRouter } from "next/router";
+import { DeleteUserDTO } from "pages/api/admin/users/delete";
+import Combobox from "components/Combobox";
+import { User } from "@prisma/client";
+import Button from "components/Button";
+// import { NextPageContext } from "next";
+import { UpdateUserDTO } from "pages/api/admin/users/update";
+import { UserRoleType, IUser } from "interfaces";
+
+const UserAccountPage: React.FunctionComponent<UserRequest> = ({
+  onDelete,
+  onAccept,
+}) => {
+  const router = useRouter();
+  const [selectedPlayers, setSelectedPlayers] = useState<User[]>([]);
+  // const [user, setUser] = useState<User[]>([]);
+  // const [roleChosen] = useState("");
+  const [user, setUser] = useState<IUser>();
+  // const [isEditing, setIsEditing] = useState(false);
+  // const router = useRouter();
+  const { id } = router.query;
+  // const createDate = new Date().toDateString;
+  useEffect(() => {
+    const getUser = async (): Promise<void> => {
+      const response = await fetch(`/api/admin/users/${id}`, {
+        method: "GET",
+        headers: { "content-type": "application/json" },
+        redirect: "follow",
+      });
+      const data = await response.json();
+      setUser(data.user);
+    };
+    getUser();
+  }, [id]);
+
+  const showCombobox = (): string => {
+    return ([
+      UserRoleType.Mentor,
+      UserRoleType.Parent,
+      UserRoleType.Donor,
+    ] as UserRoleType[]).includes(user?.defaultRole.type as UserRoleType)
+      ? ""
+      : "hidden";
+  };
+  const deleteUser = async (): Promise<void> => {
+    try {
+      const response = await fetch("/api/admin/users/delete", {
+        method: "DELETE",
+        headers: { "Content-Type": "application/json" },
+        credentials: "include",
+        body: JSON.stringify({
+          id,
+        } as DeleteUserDTO),
+      });
+      if (!response.ok) {
+        throw await response.json();
+      }
+      onDelete();
+    } catch (err) {
+      throw new Error(err.message);
+    }
+  };
+
+  const acceptUser = async (): Promise<void> => {
+    try {
+      const response = await fetch(`/api/admin/users/${user?.id}`, {
+        method: "PATCH",
+        headers: { "Content-Type": "application/json" },
+        credentials: "include",
+        body: JSON.stringify(({
+          emailVerified: new Date(),
+          upstreamRoles: selectedPlayers,
```

Comment:
I believe that users that are not players do not have upstreamRoles (it is only for players to know which users are linked to them), which is why this line throws an error. What you actually want to do is use the user/update or role/create API endpoint to create be able to update the user with the correct players. For non-player users, the connected players are held in role not upstreamRoles

Reply:
> What you actually want to do is use the user/update

@sydbui This network request _is_ actually reaching the admin/users/update endpoint (it gets routed to it by admin/users/[id], which has a `routeByMethod` call that matches the PATCH method route to the update endpoint).

@christopher-grey `upstreamRoles` is not a field that is accepted by this endpoint. Right now, this endpoint doesn't actually meet the use case that we need here, which is to replace the roles that they already have with the newly specified roles, _if_ they have specified it as a part of the POST body.

Here's the approach I would take here:
1. We made a change in this PR to default the roles to `[]` if it's undefined. Let's remove that behavior, since `undefined` will now mean "leave roles alone."
2. If roles is _not_ undefined, we will _create_ roles with the details specified by roles, coming from the request body. For now, we'll have to make this be a series of `prisma.role.create` calls, as there is no first-class "createMany" function until Prisma 2.16 (which we don't have yet).

Unfortunately making this change will also necessitate an update to the page that currently uses this endpoint, which is the "edit a user" view where we allow them to switch roles. After a conversation with Rachel, it seems that this page is actually missing the Combobox (preventing the admins from making changes to Linked player lists at the moment).

My suggestion for the course of action here: let's nix `upstreamRoles` for now:
```suggestion
```

and file a new card for updating the behavior I mentioned above, which can be summed up as: "Update users endpoint should allow for full role detail replacement".

The work involved will include both the API endpoint update change and the display changes required (in this page for the network request and in the admin/user/[id] page).